lib/jekyll-relative-links/generator.rb

```diff
@@ -21,16 +21,52 @@ def initialize(site)
     end
 
     def generate(site)
+      # Get any configuration specified
+      # Defaults mean no change in behaviour
+      jrl_config = site.config["jekyll_relative_links"] || {}
+      @config = {}
+      @config["collections"] = jrl_config["collections"] || []
+      @config["process_all_collections"] = jrl_config["process_all_collections"] || false
+      @config["verbose"] = jrl_config["verbose"] || 0
+
       @site    = site
       @context = context
 
       site.pages.each do |page|
         next unless markdown_extension?(page.extname)
         replace_relative_links!(page)
       end
+
+      site.collections.each do |collection|
```

Comment:
I was trying to make minimal impact on your original code :). My immediate thought about your suggestion, though, is that you are then iterating through things twice. The first pass collects a list of pages to process and the second pass then processes that list ... unless I've misunderstood your suggestion?